kodi_addon_checker/check_addon.py

```diff
@@ -435,3 +389,23 @@ def _check_dependencies(report: Report, addon_path, repo_addons):
             if LooseVersion(available_version) < LooseVersion(required_version) and (required_addon not in ignore):
                 report.add(Record(PROBLEM, "Version mismatch for addon %s. Required: %s, Available: %s "
                                   % (required_addon, required_version, available_version)))
+
+
+def get_addon_name(xml_path):
+    tree = ET.parse(xml_path).getroot()
+    return (tree.get("id"), tree.get("version"))
+
+
+def _check_for_existing_addon(report: Report, addon_xml, repo_addons):
+    """Check if addon submitted already exists or not"""
+    addon_xml = os.path.join(addon_xml, "addon.xml")
+    addon_name, addon_version = get_addon_name(addon_xml)
+
+    if addon_name not in repo_addons:
+        report.add(Record(WARNING, "This is a new addon"))
+    else:
+        report.add(Record(WARNING, "%s addon already exist with version %s."
```

Comment:
Notice here we are not printing the branch name that is because we get the repo addon directly from the __main__ file, that was done to cache the addon.xml file.

Reply:
This shouldn't be a warning (only info), as it only checks if the addon is already in the repo which it is PRed to.